pkg/kube/namespace/filter.go

```diff
@@ -79,8 +79,13 @@ func NewDiscoveryNamespacesFilter(
 			}
 		},
 		DeleteFunc: func(ns *corev1.Namespace) {
-			// No need to notify handlers for deletes
-			f.namespaceDeleted(ns.ObjectMeta)
+			f.lock.Lock()
+			defer f.lock.Unlock()
```

Comment:
LGTM, I think this is the critical fix.

But what is the target for updating  pkg/kube/kclient/client.go